File: src-tauri/src/cli/detector/languages.rs
use crate::cli::detector::ProjectTemplate;
use std::fs;
use std::path::Path;

/// Detect Docker projects
pub fn detect_docker(directory: &Path) -> Option<Vec<ProjectTemplate>> {
    let dockerfile_path = directory.join("Dockerfile");

    if !dockerfile_path.exists() {
        return None;
    }

    let projects = vec![ProjectTemplate {
        name: "docker: build".to_string(),
        command: "docker build -t $(basename $(pwd)) .".to_string(),
        description: "Build Docker image".to_string(),
    }];

    Some(projects)
}

/// Detect Python projects
pub fn detect_python(directory: &Path) -> Vec<ProjectTemplate> {
    let mut projects = Vec::new();

    // Check for requirements.txt
    if directory.join("requirements.txt").exists() {
        projects.push(ProjectTemplate {
            name: "pip: install".to_string(),
            command: "pip install -r requirements.txt".to_string(),
            description: "Install Python dependencies".to_string(),
        });
    }

    // Check for poetry
    if directory.join("pyproject.toml").exists() {
        projects.push(ProjectTemplate {
            name: "poetry: install".to_string(),
            command: "poetry install".to_string(),
            description: "Install Poetry dependencies".to_string(),
        });
    }

    // Check for pipenv
    if directory.join("Pipfile").exists() {
        projects.push(ProjectTemplate {
            name: "pipenv: install".to_string(),
            command: "pipenv install".to_string(),
            description: "Install Pipenv dependencies".to_string(),
        });
    }

    projects
}

/// Detect Rust projects
pub fn detect_rust(directory: &Path) -> Vec<ProjectTemplate> {
    let mut projects = Vec::new();

    if directory.join("Cargo.toml").exists() {
        projects.push(ProjectTemplate {
            name: "cargo: build".to_string(),
            command: "cargo build".to_string(),
            description: "Build Rust project".to_string(),
        });

        projects.push(ProjectTemplate {
            name: "cargo: test".to_string(),
            command: "cargo test".to_string(),
            description: "Run Rust tests".to_string(),
        });

        // Check if it's a workspace
        if let Ok(content) = fs::read_to_string(directory.join("Cargo.toml")) {
            if content.contains("[workspace]") {
                projects.push(ProjectTemplate {
                    name: "cargo: build --workspace".to_string(),
                    command: "cargo build --workspace".to_string(),
                    description: "Build entire workspace".to_string(),
                });
            }
        }
    }

    projects
}

/// Detect Go projects
pub fn detect_go(directory: &Path) -> Vec<ProjectTemplate> {
    let mut projects = Vec::new();

    if directory.join("go.mod").exists() {
        projects.push(ProjectTemplate {
            name: "go: build".to_string(),
            command: "go build ./...".to_string(),
            description: "Build Go project".to_string(),
        });

        projects.push(ProjectTemplate {
            name: "go: test".to_string(),
            command: "go test ./...".to_string(),
            description: "Run Go tests".to_string(),
        });

        projects.push(ProjectTemplate {
            name: "go: run".to_string(),
            command: "go run .".to_string(),
            description: "Run Go application".to_string(),
        });
    }

    projects
}

File: src-electron/cli/detector/languages.ts
/**
 * Language-specific project detectors (Docker, Python, Rust, Go)
 */

import * as fs from 'fs';
import * as path from 'path';
import type { ProjectTemplate } from '../types';

export function detectDocker(directory: string): ProjectTemplate[] | null {
  if (!fs.existsSync(path.join(directory, 'Dockerfile'))) {
    return null;
  }

  return [{
    name: 'docker: build',
    command: 'docker build -t $(basename $(pwd)) .',
    description: 'Build Docker image',
    priority: 30
  }];
}

export function detectPython(directory: string): ProjectTemplate[] {
  const projects: ProjectTemplate[] = [];

  if (fs.existsSync(path.join(directory, 'requirements.txt'))) {
    projects.push({
      name: 'pip: install',
      command: 'pip install -r requirements.txt',
      description: 'Install Python dependencies',
      priority: 31
    });
  }

  if (fs.existsSync(path.join(directory, 'pyproject.toml'))) {
    projects.push({
      name: 'poetry: install',
      command: 'poetry install',
      description: 'Install Poetry dependencies',
      priority: 32
    });
  }

  if (fs.existsSync(path.join(directory, 'Pipfile'))) {
    projects.push({
      name: 'pipenv: install',
      command: 'pipenv install',
      description: 'Install Pipenv dependencies',
      priority: 33
    });
  }

  return projects;
}

export function detectRust(directory: string): ProjectTemplate[] {
  const projects: ProjectTemplate[] = [];
  const cargoPath = path.join(directory, 'Cargo.toml');

  if (!fs.existsSync(cargoPath)) {
    return projects;
  }

  projects.push({
    name: 'cargo: build',
    command: 'cargo build',
    description: 'Build Rust project',
    priority: 34
  });

  projects.push({
    name: 'cargo: test',
    command: 'cargo test',
    description: 'Run Rust tests',
    priority: 35
  });

  // Check if it's a workspace
  try {
    const content = fs.readFileSync(cargoPath, 'utf-8');
    if (content.includes('[workspace]')) {
      projects.push({
        name: 'cargo: build --workspace',
        command: 'cargo build --workspace',
        description: 'Build entire workspace',
        priority: 36
      });
    }
  } catch {
    // Ignore read errors
  }

  return projects;
}

export function detectGo(directory: string): ProjectTemplate[] {
  const projects: ProjectTemplate[] = [];

  if (!fs.existsSync(path.join(directory, 'go.mod'))) {
    return projects;
  }

  projects.push({
    name: 'go: build',
    command: 'go build ./...',
    description: 'Build Go project',
    priority: 37
  });

  projects.push({
    name: 'go: test',
    command: 'go test ./...',
    description: 'Run Go tests',
    priority: 38
  });

  projects.push({
    name: 'go: run',
    command: 'go run .',
    description: 'Run Go application',
    priority: 39
  });

  return projects;
}
